apps/api/src/app/inbox/usecases/update-notification/update-notification.usecase.ts

```diff
@@ -0,0 +1,133 @@
+      _environmentId: command.environmentId,
+      _subscriberId: subscriber._id,
+      _id: command.notificationId,
+    });
+    if (!message) {
+      throw new NotFoundException(`Notification with id: ${command.notificationId} is not found.`);
+    }
+
+    const isUpdatingRead = command.read !== undefined;
+    const isUpdatingArchived = command.archived !== undefined;
+    const isUpdatingPrimaryCta = command.primaryActionCompleted !== undefined;
+    const isUpdatingSecondaryCta = command.secondaryActionCompleted !== undefined;
+    const primaryCta = message.cta.action?.buttons?.find((button) => button.type === ButtonTypeEnum.PRIMARY);
+    const secondaryCta = message.cta.action?.buttons?.find((button) => button.type === ButtonTypeEnum.SECONDARY);
+    if ((isUpdatingPrimaryCta && !primaryCta) || (isUpdatingSecondaryCta && !secondaryCta)) {
+      throw new ApiException(
+        `Could not perform action on the ${isUpdatingPrimaryCta && !primaryCta ? 'primary' : 'secondary'} button.`
+      );
+    }
+
+    const updatePayload: Partial<MessageEntity> = {
+      ...(isUpdatingRead && { read: command.read }),
+      ...(isUpdatingArchived && { archived: command.archived }),
+    };
+
+    if (isUpdatingPrimaryCta) {
+      updatePayload['cta.action.result.type'] = ButtonTypeEnum.PRIMARY;
+      updatePayload['cta.action.status'] = command.primaryActionCompleted
+        ? MessageActionStatusEnum.DONE
+        : MessageActionStatusEnum.PENDING;
+    }
+
+    if (isUpdatingSecondaryCta) {
+      updatePayload['cta.action.result.type'] = ButtonTypeEnum.SECONDARY;
+      updatePayload['cta.action.status'] = command.secondaryActionCompleted
+        ? MessageActionStatusEnum.DONE
+        : MessageActionStatusEnum.PENDING;
+    }
+
+    await this.messageRepository.update(
+      {
+        _environmentId: command.environmentId,
+        _subscriberId: subscriber._id,
+        _id: command.notificationId,
+      },
+      {
+        $set: updatePayload,
+      }
+    );
+
+    await this.invalidateCache.invalidateQuery({
+      key: buildFeedKey().invalidate({
+        subscriberId: subscriber.subscriberId,
+        _environmentId: command.environmentId,
+      }),
+    });
+
+    await this.invalidateCache.invalidateQuery({
+      key: buildMessageCountKey().invalidate({
+        subscriberId: subscriber.subscriberId,
+        _environmentId: command.environmentId,
+      }),
+    });
+
+    this.analyticsService.mixpanelTrack(AnalyticsEventsEnum.UPDATE_NOTIFICATION, '', {
+      _organization: command.organizationId,
+      _subscriber: subscriber._id,
+      _notification: command.notificationId,
+      read: command.read,
+      archived: command.archived,
+      primaryActionCompleted: command.primaryActionCompleted,
+      secondaryActionCompleted: command.secondaryActionCompleted,
+    });
+
+    if (isUpdatingRead) {
+      this.webSocketsQueueService.add({
+        name: 'sendMessage',
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Should it be `sendMessage`? 🤔 ‏

Reply:
it's like that everywhere :) this name is used as the job name 🤷‍♂️